java/ql/lib/semmle/code/java/dataflow/internal/DataFlowNodes.qll

```diff
@@ -133,6 +135,8 @@ module Public {
       result = this.(CaptureNode).getTypeImpl()
       or
       result = this.(FieldValueNode).getField().getType()
+      or
+      result instanceof TypeObject and this instanceof AdditionalNode
```

Comment:
I guess `instanceof TypeObject` is the most restrictive assumption we can make about `result` in this case, right?

Reply:
Yes. I was thinking about whether the additional-node extension point ought to include some option to set the type, but that's probably just an unnecessary complication, since type strengthening will pick up a stronger type quickly from the surrounding nodes.